lib/improver/tests/utilities/test_mathematical_operations.py

```diff
@@ -0,0 +1,318 @@
+        coord_name = "height"
+        direction = "positive"
+        cube = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).ensure_monotonic_increase_in_chosen_direction(
+                    self.descending_cube))
+        self.assertIsInstance(cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            cube.coord(coord_name).points, self.ascending_height_points)
+
+    def test_descending_coordinate_negative(self):
+        """Test that for a monotonically ascending coordinate, where the
+        chosen direction is negative, the resulting coordinate still decreases
+        monotonically in the positive direction."""
+        coord_name = "height"
+        direction = "negative"
+        cube = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).ensure_monotonic_increase_in_chosen_direction(
+                    self.descending_cube))
+        self.assertIsInstance(cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            cube.coord(coord_name).points, self.descending_height_points)
+
+
+class Test_process(IrisTest):
+
+    """Test the process method."""
+
+    def setUp(self):
+        """Set up the cube."""
+        self.height_points = np.array([5., 10., 20.])
+        cube = set_up_height_cube(self.height_points)[:, 0, :, :, :]
+        data = np.ones(cube.shape)
+        data[1] = np.zeros(cube[1].shape)+2
+        data[2] = np.zeros(cube[2].shape)+3
+        data[0, :, 0, 0] = 6
+        cube.data = data
+        self.cube = cube
+
+    def test_basic(self):
+        """Test that a cube with the points on the chosen coordinate are
+        in the expected order."""
+        coord_name = "height"
+        direction = "negative"
+        result = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).process(self.cube))
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            result.coord("height").points, np.array([10., 5.]))
+
+    def test_data(self):
+        """Test that the resulting cube contains the expected data following
+        vertical integration."""
+        expected = np.array(
+            [[[[25.00, 25.00, 25.00],
+               [25.00, 25.00, 25.00],
+               [25.00, 25.00, 25.00]]],
+             [[[45.00, 32.50, 32.50],
+               [32.50, 32.50, 32.50],
+               [32.50, 32.50, 32.50]]]])
+        coord_name = "height"
+        direction = "negative"
+        result = (
+            Integration(
+                coord_name, direction_of_integration=direction
+                ).process(self.cube))
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(
+            result.coord("height").points, np.array([10., 5.]))
+        self.assertArrayAlmostEqual(result.data, expected)
```

Comment:
Only the last of these three assert statements belongs in this unit test. The first two were performed by test_basic.
This is true for the remaining tests too...

Reply:
I've removed the test on whether the output is a cube or not, although I've left the check on the coordinate points, as I found it useful for checking that the integration was going in the right direction.